code/api/DeploynautAPIFormatter.php

```diff
@@ -117,31 +117,43 @@ public function getDeploymentData(\DNDeployment $deployment) {
 
 	/**
 	 * Return data about a particular {@link Member} of the stack for use in API response.
-	 * Note that role can be null in the response. This is the case of an admin, or an operations
+
+	 * Notes:
+	 * 1) This method returns null instead of an array if the member doesn't exists anymore
+	 * 2) 'role' can be null in the response. This is the case of an admin, or an operations
 	 * user who can create the deployment but is not part of the stack roles.
 	 *
 	 * @param \DNProject $project
-	 * @param \Member $member
-	 * @return array
+	 * @param int $memberID
+	 * @return null|array
 	 */
-	public function getStackMemberData(\DNProject $project, \Member $member) {
+	public function getStackMemberData(\DNProject $project, $memberID) {
```

Comment:
by passing the MemberID here we don't need to initialise a full Member Object if we cached it further down